pcsx2-qt/MainWindow.cpp

```diff
@@ -1014,6 +1025,85 @@ void MainWindow::onLoggingOptionChanged()
 	QtHost::UpdateLogging();
 }
 
+void MainWindow::onInputRecNewActionTriggered()
+{
+	const bool wasPaused = VMManager::GetState() == VMState::Paused;
```

Comment:
Not thread-safe; use `m_vm_valid` and  `m_vm_paused`.

Reply:
What does `m_vm_valid` correspond to?  Looks like `true` corresponds to the vm being started/starting -- but is that the intended useage in the long-term?

Reply:
They're set when the state changes, based on a signal emitted by the VM thread. That way, you don't need to read emu thread state, which can risk changing at any time, and opens you to race conditions.

`m_vm_valid` means that a VM exists (is initialized), but any actual VMManager calls should check that it's still valid when the code actually runs on the VM thread (see the wrapper methods in EmuThread which forward to VMManager).
`m_vm_paused` is what the title says; it's paused at the time of call. Remember that these are set via signals, so at the time you check it, the VM thread may have already paused (and the UI thread hasn't processed the signal to update the bool yet).